packages/headless/src/features/redirection/redirection-slice.ts

```diff
@@ -1,11 +1,24 @@
 import {createReducer} from '@reduxjs/toolkit';
+import {executeSearch} from '../search/search-actions';
 import {checkForRedirection} from './redirection-actions';
 import {getRedirectionInitialState} from './redirection-state';
+import {
+  Trigger,
+  TriggerRedirect,
+  isTriggerRedirect,
+} from './../../api/search/trigger';
 
 export const redirectionReducer = createReducer(
   getRedirectionInitialState(),
   (builder) =>
-    builder.addCase(checkForRedirection.fulfilled, (state, action) => {
-      state.redirectTo = action.payload;
-    })
+    builder
+      .addCase(checkForRedirection.fulfilled, (state, action) => {
+        state.redirectTo = action.payload;
+      })
+      .addCase(executeSearch.fulfilled, (state, action) => {
+        const redirectTriggers: Trigger[] = action.payload.response.triggers.filter(
+          (trigger) => isTriggerRedirect(trigger)
+        );
+        state.redirectTo = (redirectTriggers[0] as TriggerRedirect).content;
```

Comment:
I wasn't sure how to handle the fact that `response.triggers` is an array of `Trigger` objects and that there might be multiple `TriggerRedirect` objects in the array.

Reply:
It's good to filter, be careful thought, because not all responses contain triggers, even less redirect triggers